pkg/virt-controller/watch/vm.go

```diff
@@ -2655,6 +2661,24 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 	return vm, syncErr, nil
 }
 
+func (c *VMController) applyDynamicIfaceRequestOnVMI(vmi *virtv1.VirtualMachineInstance, requests []virtv1.VirtualMachineInterfaceRequest, vmIfaces map[string]virtv1.Interface) error {
```

Comment:
The VM is treated in this manner: Mutate the local object through many helpers, then update the object once at the end.

Why can't we follow the same for the VMI here? I.e. mutate it as before, but this time persist the result in a vmiCopy. Then, just call the `c.vmiInterfacesPatch(vmCopy, vmi)` immediately after the update.